financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/dcr/validators/RedirectUriMatchValidator.java

```diff
@@ -0,0 +1,104 @@
+public class RedirectUriMatchValidator implements DynamicClientRegistrationValidator {
+
+    private static final Log log = LogFactory.getLog(RedirectUriMatchValidator.class);
+
+    @Override
+    public void validatePost(ApplicationRegistrationRequest applicationRegistrationRequest,
+                             Map<String, Object> ssaParams) throws FinancialServicesException {
+
+        validateRedirectUrls(applicationRegistrationRequest.getRedirectUris(), ssaParams);
+    }
+
+    @Override
+    public void validateGet(Map<String, String> ssaParams) throws FinancialServicesException {
+
+    }
+
+    @Override
+    public void validateUpdate(ApplicationUpdateRequest applicationUpdateRequest, Map<String, Object> ssaParams,
+                               ServiceProviderProperty[] serviceProviderProperties) throws FinancialServicesException {
+
+        validateRedirectUrls(applicationUpdateRequest.getRedirectUris(), ssaParams);
+    }
+
+    /**
+     * Validate the redirect URIs in the request.
+     *
+     * @param redirectUris   Redirect URIs in the request.
+     * @param ssaParams      SSA parameters.
+     * @throws FinancialServicesException When an error occurs while validating the request.
+     */
+    public static void validateRedirectUrls(List<String> redirectUris, Map<String, Object> ssaParams)
+            throws FinancialServicesException {
+        if (redirectUris != null && !redirectUris.isEmpty()) {
+
+            AtomicReference<Object> redirectURISoftwareStatement = new AtomicReference<>();
+            ssaParams.keySet().stream().forEach(key -> {
+                if (key.contains(IdentityCommonConstants.SSA_REDIRECT_URIS)) {
+                    redirectURISoftwareStatement.set(ssaParams.get(key));
+                }
+            });
+
+            if (!matchRedirectURI(redirectUris, redirectURISoftwareStatement.get())) {
+                log.error("Redirect URIs do not match with the software statement");
+                throw new FinancialServicesException("Redirect URIs do not match with the software statement");
+            }
+        }
+    }
+
+    /**
+     * Check whether the redirect uris in the request are a subset of the redirect uris in the software statement
+     * assertion.
+     */
+    public static boolean matchRedirectURI(List<String> redirectURIRequest, Object redirectURISoftwareStatement) {
+
+        int matchedURis = 0;
+        if (redirectURISoftwareStatement instanceof List) {
+            List callbackUrisSoftwareStatementValues = (List) redirectURISoftwareStatement;
+            for (String requestURI : redirectURIRequest) {
+                for (Object callbackUrisSoftwareStatementObject : callbackUrisSoftwareStatementValues) {
+                    String softwareStatementURI = (String) callbackUrisSoftwareStatementObject;
+                    if (requestURI.equals(softwareStatementURI)) {
+                        matchedURis = matchedURis + 1;
+                    }
+                }
```

Comment:
Can't we do this here?
```suggestion
                if (!callbackUrisSoftwareStatementValues.contains(requestURI)) {
                    return false;
                }
```

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/327/commits/9bdd95d9a4517faed8c734812dfcd9065d731ce0